react.json

```diff
@@ -0,0 +1,78 @@
+{
+    "extends": [
+        "plugin:import/react"
+    ],
+    "env": {
+        "browser": true
+    },
+    "plugins": [
+        "react",
+        "react-hooks"
+    ],
+    "rules": {
+        "react/destructuring-assignment": "error",
+        "react/display-name": "off",
+        "react/jsx-closing-bracket-location": [
+            "error",
+            "line-aligned"
+        ],
+        "react/jsx-curly-newline": [
+            "error",
+            {
+                "multiline": "consistent",
+                "singleline": "consistent"
+            }
+        ],
+        "react/jsx-curly-spacing": [
+            "error",
+            {
+                "children": true,
+                "when": "always"
```

Comment:
I would prefer if we change this to `never` instead of `always`. It conflicts with prettier and seems like the overall community is a bit more consistent with not using curly spacing. What do you guys think?

Reply:
I prefer to have a space around `{}` because I personally think it's more readable. Also, it is consistent with the spacing rules we have for objects i.e.: `{ foobar: 'foobar' }` and import statements i.e.: `import { foobar } from './foobar';`. But really it's a personal / team's preference. Therefore I don't mind having this rule as `never` or `always` by default. And depending on the team's preference, your project could extend this config and override the default value. Nevertheless as a company we still need to decide on the default value.

PS: Now days I have a more relaxed opinion about project overrides, especially knowing that a lot of these rules have an auto-fixer so our editor can fix the issues for us. Having said that, in general, we should stick with the defaults. But if there's a good reason and the team's happy with the overrides, I think it's okay to have them. If you can convince enough people, these overrides might eventually end up here and be used as the defaults.

Reply:
I agree that whatever we decide, some teams will still override it to match their preference. With that said, let's pick the option that most teams use. A quick GitHub Org search suggests we mostly use `never`.

Reply:
Ok cool, let's make it `never` as the default based on that fact.